### DISCUSSION TITLE
Extension of Point support to beams

### DISCUSSION THREAD

**Discussion by MartinSalak** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#232] [Ideas]
Hello SAF community, please have a look at our proposal on how to extend point supports to beams (not only nodes). All remarks are welcome.

Existing object: **StructuralPointSupport**
Currently we can assign point support only to nodes. Please have a look at our proposal for extension of this object. It aims to expand StructuralPointSupport so it can be assigned not only to nodes but to beams as well.

![image](https://user-images.githubusercontent.com/89633938/195856002-d63a2892-7332-4a28-b799-a306d90680e1.png)
 

New rows:
| Property | Example | Definition |
| --- | --- | --- |
| Boundary condition | On beam, In node | Specifies which type of object the support is assigned to |
| 1D member | B1 | The name of the StructuralCurveMember on which the support is assigned |
| Coordinate system | Global, Local | Applicable only for Boundary condition = On beam. Defines the coordinate system used by the support. Local coordinate system is defined by 1D member |
| Origin | From start, From end | Specifies where the origin for the position x coordinate measurement is |
| Coordinate definition | Absolute, Relative | Specifies the definition of the position. |
| Position x [m] | 5.25 (absolute value [m]), 0.5 (relative value [-]) | Defines the position of the support on the 1D member in local coordinate system in relative or in absolute coordinates [m] |

Changed rows:
| Property | Example | Definition |
| --- | --- | --- |
| Node | N5 | Now only required when Boundary condition = In node |

Would you like to see a different approach? Let us know what you think.

**Comment by dschuermans** [comment]
So this is pretty much the same approach as with `StructuralPointAction` I suppose?
Then I can't immediately see any issues.
